server/app/Models/User.php

```diff
@@ -448,7 +445,7 @@ public function hasRole($role)
      */
     private function beforeSave(array $options = array())
     {
-        $this->rules = $this->mergeRules();
+        static::$rules = $this->mergeRules();
```

Comment:
If you use `$this->` here, the parent class interprets `rules` to be a part of the model itself, resulting in `save()` attempting to insert a field called `rules` into the database. Obviously that doesn't exist so it breaks.